crates/wasm-sdk/src/host_calls.rs

```diff
@@ -0,0 +1,508 @@
+use {
+    std::{net::IpAddr, str, str::FromStr},
+    validator::{Validate, ValidationErrors},
+};
+
+// For some reason, doc-tests in this module trigger a linker error, so they're set to no_run
+
+use crate::bulwark_host::DecisionInterface;
+
+pub use crate::{Decision, Outcome, ParseCounterError};
+pub use http::{Extensions, Method, StatusCode, Uri, Version};
+pub use serde_json::{Map, Value};
+
+/// An HTTP request combines a head consisting of a [`Method`], [`Uri`], and headers with a [`BodyChunk`], which provides
+/// access to the first chunk of a request body.
+pub type Request = http::Request<BodyChunk>;
+/// An HTTP response combines a head consisting of a [`StatusCode`] and headers with a [`BodyChunk`], which provides
+/// access to the first chunk of a response body.
+pub type Response = http::Response<BodyChunk>;
+
+// NOTE: fields are documented via Markdown instead of normal rustdoc because the underlying type is from the macro.
```

Comment:
`lib.rs:5`

Reply:
Because `rustdoc` reads comments from above each field, but these fields are generated by macro, there's no way to emit the appropriate `rustdoc` as far as I can tell.